docs/user-manuals/workload/daemon-pod-updater.md

```diff
@@ -0,0 +1,223 @@
+```
+
+- Get pods: daemonset pods on all nodes have been upgraded
+```shell
+nginx-daemonset-kbkf6   1/1     Running   0          88s   10.244.3.3   openyurt-e2e-test-worker2   <none>           <none>
+nginx-daemonset-scgtv   1/1     Running   0          51s   10.244.2.3   openyurt-e2e-test-worker3   <none>           <none>
+nginx-daemonset-slp5t   1/1     Running   0          11m   10.244.1.3   openyurt-e2e-test-worker    <none>           <none>
+```
+
+- Check pods image version: all pods have been upgraded to nginx:1.19.5
+```shell
+***
+Containers:
+  nginx:
+    Container ID:   containerd://f7d4b3f1257a0d1d8da862671c11cb094f9fba1ba0041b7a5f783d9c9e4d8449
+    Image:          nginx:1.19.5
+    Image ID:       docker.io/library/nginx@sha256:31de7d2fd0e751685e57339d2b4a4aa175aea922e592d36a7078d72db0a45639
+    Port:           <none>
+    Host Port:      <none>
+    State:          Running
+      Started:      Fri, 14 Oct 2022 14:21:25 +0800
+    Ready:          True
+    Restart Count:  0
+    Environment:    <none>
+    Mounts:
+      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-wrhj8 (ro)
+***
+```
+
+### 3）OTA Upgrade Model
+- Create daemonset instance
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nginx-daemonset
+  annotations:
+    apps.openyurt.io/update-strategy: ota
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  updateStrategy:
+    type: OnDelete
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.19.4
+
+EOF
+
+# get nginx-daemonset pods
+$ kubectl get pods -o wide | grep nginx-daemonset
+nginx-daemonset-bwzss   1/1     Running   0          92s   10.244.3.4   openyurt-e2e-test-worker2   <none>           <none>
+nginx-daemonset-ppf9p   1/1     Running   0          92s   10.244.1.4   openyurt-e2e-test-worker    <none>           <none>
+nginx-daemonset-rgp9h   1/1     Running   0          92s   10.244.2.4   openyurt-e2e-test-worker3   <none>           <none>
+```
+
+- Check pod annotation `PodNeedUpgrade`: take pod `nginx-daemonset-bwzss` on node `openyurt-e2e-test-worker2` as an example
+```shell
+$ kubectl describe pods nginx-daemonset-bwzss
+
+***
+Conditions:
+  Type              Status
+  PodNeedUpgrade    False
+***
+```
+
+- Update daemonset: change the container image from nginx:1.19.4 to nginx:1.19.5
+```shell
+***
+containers:
+      - name: nginx
+        image: nginx:1.19.5
+***
+```
+
+- Check pod annotation `PodNeedUpgrade` again
+```shell
+$ kubectl describe pods nginx-daemonset-bwzss
+
+***
+Conditions:
+  Type              Status
+  PodNeedUpgrade    True
+***
+```
+
+- Execute OTA upgrade
+```shell
+# call Upgrade API
+$ curl -X POST 127.0.0.1:10267/openyurt.io/v1/namespaces/default/pods/nginx-daemonset-bwzss/upgrade
```

Comment:
end user should execute this command on edge node that wait for upgrading. we should add this precondition for command if end user don't known where to execute command.